src/main.rs

```diff
@@ -1,135 +1,19 @@
-            "",
-            "commit-equals",
-            "Filter for certain commit hashes. ORs if specified multiple times.",
-            "HASH",
-        )
-        .optmulti(
-            "f",
-            "file-extension",
-            "Filter loc for certain file extension (e.g. `--file-extension cpp`). ORs if specified multiple times.",
-            "EXTENSION",
-        )
-        .optmulti(
-            "m",
-            "message-contains",
-            "Filter for certain commit messages. ORs if specified multiple times.",
-            "MESSAGE",
-        )
-        .optmulti(
-            "",
-            "message-equals",
-            "Filter for certain commit messages. ORs if specified multiple times.",
-            "MESSAGE",
-        )
-        .optmulti(
-            "l",
-            "message-starts-with",
-            "Filter for certain commit messages. ORs if specified multiple times.",
-            "MESSAGE",
-        )
-        .optopt(
-            "d",
-            "duration",
-            "The time which may pass between two commits that still counts as working.",
-            "HOURS",
-        )
-        .optopt("i", "input", "The log file to read from.", "FILE")
-        .optopt(
-            "o",
-            "output",
-            "An output file for the commits per day in CSV format.",
-            "FILE",
-        );
-    let matches = match opts.parse(std::env::args()) {
-        Ok(it) => it,
-        Err(getopts::Fail::ArgumentMissing(string)) => {
-            eprintln!(
-                "{}",
-                opts.usage(&format!(
-                    "There was an argument expected for option '{string}'."
-                ))
-            );
-            return sysexits::ExitCode::Usage;
-        }
-        Err(getopts::Fail::UnrecognizedOption(string)) => {
-            eprintln!("{}", opts.usage(&format!("Unknown option '{string}'.")));
-            return sysexits::ExitCode::Usage;
-        }
-        Err(err) => {
-            eprintln!("{}", opts.usage(&format!("{err}")));
-            return sysexits::ExitCode::Config;
-        }
-    };
-    if matches.opt_present("help") {
-        commit_analyzer::usage(opts);
-        return sysexits::ExitCode::Ok;
-    }
-    let input = match commit_analyzer::InputMethod::parse(&matches) {
-        Some(input) => input,
-        None => {
-            eprintln!("{}", opts.usage("Please specify one input method."));
-            return sysexits::ExitCode::Usage;
-        }
-    };
-    let is_verbose = matches.opt_present("verbose");
-    let max_diff_hours = match matches.opt_str("duration").map(|str| str.parse::<u32>()) {
-        None => 3,
-        Some(Ok(it)) => it,
-        Some(Err(error)) => {
-            eprintln!("Invalid duration: {error}!");
-            return sysexits::ExitCode::Usage;
-        }
-    };
-    let path = matches.opt_str("output");
-    let commits = match input.read() {
+    let mut args = commit_analyzer::Args::parse();
```

Comment:
Just for completeness:  how you made the application show the help immediately when no subcommand was passed?  Is there any Clap trick to do so implicitly?

Reply:
I just copied the examples. Maybe this works, because this application has subcommands, but I do not know.. 😅